### DISCUSSION TITLE
Is there a way to use workload identity (or managed identity) to authenticate OTEL against Azure Application Insights?

### DISCUSSION THREAD

**Discussion by joshuadmatthews** [open-telemetry/opentelemetry-collector-contrib#34270] [General]
I have a policy that requires local authentication to be disabled for Application Insights. With that disabled, the instrumentation key alone will not allow access, I need to ensure calls are coming from an identity with Monitoring Metrics Publisher role. Is there any way to accomplish this?

**Comment by brunobat** [comment]
CC @jeanbisutti

**Comment by jeanbisutti** [comment]
@joshuadmatthews 

You could perhaps configure the connection string at runtime and do some verifications at this step: https://learn.microsoft.com/en-us/azure/azure-monitor/app/java-standalone-config#connection-string-configured-at-runtime 

If you need more assistance, could you please create a Github issue in https://github.com/microsoft/ApplicationInsights-Java?

**Comment by joshuadmatthews** [comment]
I don’t think the connection string is sufficient, managed identity would require fetching tokens from Entra and attaching them to the telemetry requests.

**Comment by jeanbisutti** [comment]
@joshuadmatthews 

It's perhaps what you need: https://learn.microsoft.com/en-us/azure/azure-monitor/app/azure-ad-authentication?tabs=java

**Comment by joshuadmatthews** [comment]
Yes, but I need OTEL to support that.

**Comment by jeanbisutti** [comment]
@joshuadmatthews 

It's perhaps what you are looking for: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/azuremonitorexporter/AUTHENTICATION.md#aadentra-authentication
